app/src/lib/hostedServices/github/githubService.ts

```diff
@@ -0,0 +1,35 @@
+import { GitHubChecksMonitor } from './githubChecksMonitor';
+import { GitHubListingService } from './githubListingService';
+import { GitHubPrService } from './githubPrService';
+import { Octokit } from '@octokit/rest';
+import type { ProjectMetrics } from '$lib/metrics/projectMetrics';
+import type { RepoInfo } from '$lib/url/gitUrl';
+import type { HostedGitChecksMonitor } from '../interface/hostedGitChecksMonitor';
+import type { HostedGitListingService } from '../interface/hostedGitListingService';
+import type { HostedGitPrService } from '../interface/hostedGitPrService';
+import type { HostedGitService } from '../interface/hostedGitService';
+import type { DetailedPullRequest } from '../interface/types';
+
+export type PrAction = 'creating_pr';
+export type PrState = { busy: boolean; branchId: string; action?: PrAction };
+export type PrCacheKey = { value: DetailedPullRequest | undefined; fetchedAt: Date };
+
+export class GitHubService implements HostedGitService {
+	constructor(
+		private projectMetrics: ProjectMetrics,
+		private octokit: Octokit,
+		private repo: RepoInfo
+	) {}
+
+	listService(): HostedGitListingService {
+		return new GitHubListingService(this.projectMetrics, this.octokit, this.repo);
+	}
+
+	prService(baseBranch: string, upstreamName: string): HostedGitPrService {
+		return new GitHubPrService(this.octokit, this.repo, baseBranch, upstreamName);
+	}
+
+	checksMonitor(sourceBranch: string): HostedGitChecksMonitor {
+		return new GitHubChecksMonitor(this.octokit, this.repo, sourceBranch);
+	}
```

Comment:
I don't think this makes sense

We have the DefaultHostedGitServiceFactory, which will create a specific GitHubService, which is actually a factory for these smaller services.

I don't really see the benefit of the intermediary step. The DefaultHostedGitServiceFactory would be better of being a factory for the listService, prService, and checksMonitor.

Reply:
As discussed offline, we'll keep this for now as the purpose is not having to provide `RepoInfo` and `Octokit` when constructing the individual services.